File: electrical/src/main/java/resonantinduction/electrical/tesla/RenderTesla.java
package resonantinduction.electrical.tesla;

import net.minecraft.client.renderer.tileentity.TileEntitySpecialRenderer;
import net.minecraft.tileentity.TileEntity;
import net.minecraft.util.ResourceLocation;
import net.minecraftforge.client.model.AdvancedModelLoader;
import net.minecraftforge.client.model.IModelCustom;

import org.lwjgl.opengl.GL11;

import resonantinduction.core.Reference;
import cpw.mods.fml.relauncher.Side;
import cpw.mods.fml.relauncher.SideOnly;

/**
 * @author Calclavia
 * 
 */
@SideOnly(Side.CLIENT)
public class RenderTesla extends TileEntitySpecialRenderer
{
	public static final ResourceLocation TEXTURE_BOTTOM = new ResourceLocation(Reference.DOMAIN, Reference.MODEL_PATH + "tesla_bottom.png");
	public static final ResourceLocation TEXTURE_MIDDLE = new ResourceLocation(Reference.DOMAIN, Reference.MODEL_PATH + "tesla_middle.png");
	public static final ResourceLocation TEXTURE_TOP = new ResourceLocation(Reference.DOMAIN, Reference.MODEL_PATH + "tesla_top.png");

	public static final IModelCustom MODEL_BOTTOM = AdvancedModelLoader.loadModel(Reference.MODEL_DIRECTORY + "teslaBottom.tcn");
	public static final IModelCustom MODEL_MIDDLE = AdvancedModelLoader.loadModel(Reference.MODEL_DIRECTORY + "teslaMiddle.tcn");
	public static final IModelCustom MODEL_TOP = AdvancedModelLoader.loadModel(Reference.MODEL_DIRECTORY + "teslaTop.tcn");

	@Override
	public void renderTileEntityAt(TileEntity t, double x, double y, double z, float f)
	{
		GL11.glPushMatrix();
		GL11.glTranslated(x + 0.5, y + 0.5, z + 0.5);

		int meta = t.getBlockType() != null ? t.getBlockMetadata() : 0;

		switch (meta)
		{
			default:
				bindTexture(TEXTURE_BOTTOM);
				MODEL_BOTTOM.renderAll();
				break;
			case 1:
				bindTexture(TEXTURE_MIDDLE);
				MODEL_MIDDLE.renderAll();
				break;
			case 2:
				bindTexture(TEXTURE_TOP);
				MODEL_TOP.renderAll();
				break;
		}

		GL11.glPopMatrix();
	}

}

File: archives/tesla/RenderTesla.scala
package com.calclavia.edx.electrical.circuit.component.tesla

import cpw.mods.fml.relauncher.{Side, SideOnly}
import edx.core.Reference
import net.minecraft.client.renderer.tileentity.TileEntitySpecialRenderer
import net.minecraft.tileentity.TileEntity
import net.minecraft.util.ResourceLocation
import net.minecraftforge.client.model.{AdvancedModelLoader, IModelCustom}
import org.lwjgl.opengl.GL11

/**
 * @author Calclavia
 *
 */
@SideOnly(Side.CLIENT) object RenderTesla
{
  final val TEXTURE_BOTTOM: ResourceLocation = new ResourceLocation(Reference.domain, Reference.modelPath + "tesla_bottom.png")
  final val TEXTURE_MIDDLE: ResourceLocation = new ResourceLocation(Reference.domain, Reference.modelPath + "tesla_middle.png")
  final val TEXTURE_TOP: ResourceLocation = new ResourceLocation(Reference.domain, Reference.modelPath + "tesla_top.png")
  final val MODEL_BOTTOM: IModelCustom = AdvancedModelLoader.loadModel(new ResourceLocation(Reference.domain, Reference.modelPath + "teslaBottom.tcn"))
  final val MODEL_MIDDLE: IModelCustom = AdvancedModelLoader.loadModel(new ResourceLocation(Reference.domain, Reference.modelPath + "teslaMiddle.tcn"))
  final val MODEL_TOP: IModelCustom = AdvancedModelLoader.loadModel(new ResourceLocation(Reference.domain, Reference.modelPath + "teslaTop.tcn"))
}

@SideOnly(Side.CLIENT)
class RenderTesla extends TileEntitySpecialRenderer
{
  def renderTileEntityAt(t: TileEntity, x: Double, y: Double, z: Double, f: Float)
  {
    GL11.glPushMatrix
    GL11.glTranslated(x + 0.5, y + 0.5, z + 0.5)
    val meta: Int = if (t.getBlockType != null) t.getBlockMetadata else 0

    if (meta == 1)
    {
      bindTexture(RenderTesla.TEXTURE_MIDDLE)
      RenderTesla.MODEL_MIDDLE.renderAll
    }
    else if (meta == 2)
    {
      bindTexture(RenderTesla.TEXTURE_TOP)
      RenderTesla.MODEL_TOP.renderAll
    }
    else
    {
      bindTexture(RenderTesla.TEXTURE_BOTTOM)
      RenderTesla.MODEL_BOTTOM.renderAll
    }
    GL11.glPopMatrix
  }
}